kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java

```diff
@@ -183,17 +186,23 @@ public FilteredColumnarBatch next() {
                 InternalScanFileUtils.getPartitionValues(scanFile),
                 physicalReadSchema);
 
-        DeletionVectorDescriptor dv =
-            InternalScanFileUtils.getDeletionVectorDescriptorFromRow(scanFile);
-
-        int rowIndexOrdinal =
-            nextDataBatch.getSchema().indexOf(StructField.METADATA_ROW_INDEX_COLUMN_NAME);
+        // Transform physical row tracking columns to logical row tracking columns
+        Map<String, String> configuration = ScanStateRow.getConfiguration(scanState);
+        if (TableConfig.ROW_TRACKING_ENABLED.fromMetadata(configuration)) {
```

Comment:
I'm not sure this is the right conditional? Shouldn't we never have a rowTracking column in the readSchema if it's not enabled?

Reply:
Checking if row tracking columns are enabled essentially is would the function we call here does so it would duplicate logic. 
I like your proposed comment to clarify things - I'll add it tomorrow.